contracts/hydro/src/contract.rs

```diff
@@ -831,7 +870,7 @@ fn vote(
             )?;
 
             // update the proposal in the proposal map, as well as the props by score map
-            update_proposal_and_props_by_score_maps(
+            let updated_proposal = update_proposal_and_props_by_score_maps(
```

Comment:
I was a bit iffy on this (it's using the return for something completely different to what the func is doing), but I think it's elegant, since one can reasonably expect the func to have a return, so I think it ends up being quite nice :)

Reply:
Yeah, I had the same feeling about it. Good, we can go with this approach. :)